compiler/rustc_mir_transform/src/const_prop.rs

```diff
@@ -367,7 +378,7 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 op
             }
             Err(e) => {
-                trace!("get_const failed: {}", e);
+                trace!("get_const failed: {e:?}");
```

Comment:
More suspicious debug printing. Does this regress how readable traces are?

Reply:
As someone who used these messages for debugging in the past I am certain this is worse than before -- the exact variant doesn't matter; the information about what went wrong matters. (We have RUST_CTFE_BACKTRACE=immediate to figure out where the error comes from, which is a lo more helpful than the variant anyway.) Maybe the ugly debug output is good enough but I object to attempts to selling this as an improvement.

In Miri there is some code which turns errors into a string. Is it possible to use that for a `Display` impl for `InterpError`?

Reply:
Ah I see. I could use that to print the diagnostic message if that is preferred.

Reply:
Yeah I think that would be better than debug-printing. The code can be shared with Miri, I assume.

However this can happen in a follow-up PR. Let's get this one landed before it bitrots.